issue-17: Add arguments to make the prompt shorter

reicolina opened the pull request:
This is for issue https://github.com/michaeldfallen/git-radar/issues/17. Depending of the status of your git project the prompt rendered by git-radar could get a little long.

As per discussion in https://github.com/michaeldfallen/git-radar/issues/17, adding a `--minimal` arg to hide the `git:` part, and a `--no-remote-status` arg that will hide the remote commits status part.

This PR includes:
- Main code changes, to add the args described above
- Unit tests
- An update to `README.md` to add a description for these new args

reicolina committed:
issue-17: Add arguments to make the prompt shorter

michaeldfallen commented:
@reinaldo13 nice work. Can I be a little annoying and ask you to split the two new functions apart? Thing is I really like the `--no-remote-status` part but the `--minimal` feels like it's a very narrow solution. 

I'd prefer to make the ` git:(` part configurable, perhaps using ENV vars, that way anyone can remove those parts and make it minimal through an `export GIT_RADAR_COMMITS_PREFIX=""` or something similar.

What do you think?

reinaldo13 commented:
Hi @michaeldfallen. You were not annoying at all :smile:. A lot of people are/will be using git-radar so i think changes like this deserve healthy discussions.

Since we both agree on the `--no-remote-status` arg, i will submit another commit to this PR to remove the `--minimal` arg and only include the `--no-remote-status`.

I will send a separate PR ticket to deal with the commits prefix `git:(`. I really like the approach you suggested; making it configurable will allow users to either remove it or put their own ones there. Stay tuned :radio:

reicolina committed:
Removing --minimal arg implementation

- Dealing with the git: prefix will be done in a separate pull request
- The new --no-remote-status arg remains

reinaldo13 commented:
@michaeldfallen: I just added a commit to this PR to remove the `--minimal` arg implementation and leave the `--no-remote-status` one. Feel free to merge if you find it appropriate.

I will be sending a separate PR that deals with the `git:` prefix.

Thanks again.

reicolina committed:
Added change for issue-17 to prompt.zsh as well

michaeldfallen commented:
:+1: looks good to me. Making the prefix suffixes configurable also handles https://github.com/michaeldfallen/git-radar/issues/15 so two birds :-).

Merging
